common/libs/units/units.h

```diff
@@ -110,11 +110,31 @@ template <class Q, class ValTy> class Scalar {
   ValTy val_;
 };
 
-// Represents a Scalar where addition and subtraction are well-defined.
+// Represents a Scalar where:
+// - addition and subtraction are well-defined
+// - multiplication by an external factor of type ValTy is defined
+// This is what mathematicians call a "module":
+// https://en.wikipedia.org/wiki/Module_(mathematics)
+// Note that multiplication by an external factor requires multiplication to be
+// well defined on ValTy type.
+// Be aware that dividing by 0.0f yields Inf and needs to be protected.
 template <class Q, class ValTy> class ArithScalar : public Scalar<Q, ValTy> {
 public:
   constexpr Q operator+(const ArithScalar &a) { return Q(this->val_ + a.val_); }
   constexpr Q operator-(const ArithScalar &a) { return Q(this->val_ - a.val_); }
+  inline ArithScalar &operator+=(const ArithScalar &a) {
+    return *this = *this + a;
+  }
+  inline ArithScalar &operator-=(const ArithScalar &a) {
+    return *this = *this - a;
+  }
+  constexpr Q operator*(const ValTy &a) { return Q(this->val_ * a); }
+  constexpr Q operator/(const ValTy &a) { return Q(this->val_ / a); }
+  inline ArithScalar &operator*=(const ValTy &a) { return *this = *this * a; }
```

Comment:
Don't need "inline" on functions whose definition lives inside of a class definition.  This is implicit.

Reply:
OK